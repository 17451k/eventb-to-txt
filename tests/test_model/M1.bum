<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<org.eventb.core.machineFile org.eventb.core.comment="comment" org.eventb.core.configuration="org.eventb.core.fwd" version="5">
<org.eventb.core.refinesMachine name="'" org.eventb.core.target="M0"/>
<org.eventb.core.seesContext name="-" org.eventb.core.target="C1"/>
<org.eventb.core.variable name="(" org.eventb.core.identifier="var1"/>
<org.eventb.core.variable name=")" org.eventb.core.identifier="var2"/>
<org.eventb.core.event name="." org.eventb.core.convergence="0" org.eventb.core.extended="true" org.eventb.core.label="INITIALISATION"/>
<org.eventb.core.event name="/" org.eventb.core.comment="comment" org.eventb.core.convergence="2" org.eventb.core.extended="true" org.eventb.core.label="evt1">
<org.eventb.core.refinesEvent name="'" org.eventb.core.target="evt1"/>
</org.eventb.core.event>
<org.eventb.core.event name="0" org.eventb.core.convergence="1" org.eventb.core.extended="false" org.eventb.core.label="evt2">
<org.eventb.core.refinesEvent name="'" org.eventb.core.target="evt1"/>
<org.eventb.core.parameter name="'" org.eventb.core.identifier="prm2"/>
<org.eventb.core.guard name="(" org.eventb.core.label="grd1" org.eventb.core.predicate="prm2=1"/>
<org.eventb.core.action name=")" org.eventb.core.assignment="var1≔var1−prm2" org.eventb.core.label="act1"/>
<org.eventb.core.action name="*" org.eventb.core.assignment="var2≔{1}" org.eventb.core.label="act2"/>
<org.eventb.core.guard name="-" org.eventb.core.label="grd2" org.eventb.core.predicate="var1≠prm2"/>
<org.eventb.core.guard name="," org.eventb.core.label="grd3" org.eventb.core.predicate="var1&gt;0"/>
<org.eventb.core.witness name="." org.eventb.core.comment="comment" org.eventb.core.label="prm1" org.eventb.core.predicate="prm1=prm2"/>
</org.eventb.core.event>
<org.eventb.core.event name="1" org.eventb.core.convergence="0" org.eventb.core.extended="false" org.eventb.core.label="evt3"/>
</org.eventb.core.machineFile>
