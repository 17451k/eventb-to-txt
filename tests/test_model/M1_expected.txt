machine M1 // comment
    refines M0
    sees C1

variables
    var1
    var2

events
    event INITIALISATION extends INITIALISATION
    end

    anticipated event evt1 extends evt1 // comment
    end

    convergent event evt2 refines evt1
      any
        prm2
      where
        @grd1 prm2 = 1
        @grd2 var1 ≠ prm2
        @grd3 var1 > 0 // multiline comment
      with
        @prm1 prm1 = prm2 // comment
      then
        @act1 var1 ≔ var1 − prm2
        @act2 var2 ≔ {1} // multi line comment
    end

    event evt3
    end

end
